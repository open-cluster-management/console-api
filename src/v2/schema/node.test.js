/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 * Copyright (c) 2020 Red Hat, Inc.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('Nodes Resolver', () => {
  test('Correctly Resolves Nodes Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          node(name: "hub-cluster") {
            cluster {
              clusterip
              metadata {
                name
              }
            }
            labels
            roles
            status
          }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));
});
