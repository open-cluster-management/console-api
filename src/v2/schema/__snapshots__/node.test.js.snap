// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nodes Resolver Correctly Resolves Nodes Query 1`] = `
Object {
  "data": Object {
    "node": Array [
      Object {
        "cluster": Object {
          "clusterip": "9.42.23.230",
          "metadata": Object {
            "name": "hub-cluster",
          },
        },
        "labels": Object {
          "beta.kubernetes.io/arch": "amd64",
          "beta.kubernetes.io/os": "linux",
          "kubernetes.io/hostname": "9.37.137.174",
          "node-role.kubernetes.io/worker": "true",
        },
        "roles": Array [
          "worker",
        ],
        "status": "True",
      },
      Object {
        "cluster": Object {
          "clusterip": "9.42.23.230",
          "metadata": Object {
            "name": "hub-cluster",
          },
        },
        "labels": Object {
          "beta.kubernetes.io/arch": "amd64",
          "beta.kubernetes.io/os": "linux",
          "kubernetes.io/hostname": "9.37.137.92",
          "node-role.kubernetes.io/worker": "true",
        },
        "roles": Array [
          "worker",
        ],
        "status": "True",
      },
      Object {
        "cluster": Object {
          "clusterip": "9.42.23.230",
          "metadata": Object {
            "name": "hub-cluster",
          },
        },
        "labels": Object {
          "beta.kubernetes.io/arch": "amd64",
          "beta.kubernetes.io/os": "linux",
          "etcd": "true",
          "kubernetes.io/hostname": "9.42.23.230",
          "management": "true",
          "master": "true",
          "node-role.kubernetes.io/etcd": "true",
          "node-role.kubernetes.io/management": "true",
          "node-role.kubernetes.io/master": "true",
          "node-role.kubernetes.io/proxy": "true",
          "proxy": "true",
          "role": "master",
        },
        "roles": Array [
          "etcd",
          "management",
          "master",
          "proxy",
        ],
        "status": "True",
      },
    ],
  },
}
`;
